### DISCUSSION TITLE
Meeting Minute

### DISCUSSION THREAD

**Discussion by senamihodonu** [cs481-ekh/f22-dust-devils#2] [Meeting Minutes]
[ date if not submitted on the day of the meeting ]
Excused: [ list of those not in attendance who notified the group ]
Absent: [ list of those not in attendance who did not notify the group ]

Summary: [ summarize the topics that were discussed ]

Action Items: [ list of action times by assignee ]

**Comment by lukaskarlsson290** [comment]
09/07/2022 - Sprint 0
Excused: N/A
Absent: N/A

Summary: 

- Put together Sprint 0 plan on GitHub
- Built plan in Product Backlog in preparation for work tomorrow

Action Items:

- [x] Clone repo - Sena
- [x] Setup PyTest - Sam
- [ ] Setup Linter - Lukas
- [ ] Make sure product works as expected on team side - Jim

**Comment by senamihodonu** [comment]
09/13/2022 - Sprint 1
Excused: N/A
Absent: N/A

Summary: 
- Meeting started with a sprint retroactive. It was pointed out that the main reason some of the goals were not met was due to lack of clarity on the deliverables and concluded that team will do better going forward.
- Team plans to have a meeting every Monday to fully layout team goals and tasks for the week.  
- Switched team roles for Sprint 1
  - Team Lead: Samantha
  -  Product Owner: Jim 
  -  Scribe: Sena
  -  Dev: Lukas
- Discussed epics and brainstormed on user stories
- Built plan in Product Backlog in preparation for work tomorrow

Action Items:

- [ ] Setup CI strategy for project - Sena
- [ ] Perform testing on the current project setup: Sam, Lukas
- [ ] Research PDS4 - Jim
- [ ] Make edit and complete project plan - Team

**Comment by senamihodonu** [comment]
09/20/2022 - Sprint 1
Excused: Jim Crowell
Absent: N/A

Summary:

As three of the team members were not present in class and zoom was not available class day of 9/20/2022, the meeting started with a recap of discussions and announcements from class. We then proceeded to give a status update on assigned tasks. After this we talked about the upcoming homework and ended the meeting by doing a final look through of the project plan.

Key notes:
- Team will do a better job of writing tasks for user stories
- Team will do better with code check-ins and documentation

**Comment by JimmyCIsCrazySee** [comment]
09/27/22 - Sprint 1 End/Sprint 2 Init
All present

**Summary:**
Completed Sprint 1 Retro

After taking stock of the outcomes of Sprint 1, as can be seen on that retrospective, as we were stymied on the front of including PDS4 functionality and documentation for the project, we decided to make those the focus of this next Sprint. We itemized the project requirements as **Sprint Goals**:
- PDS4 (Understanding and beginning implementation, or at least wireframing how it will be integrated)
- Wind Time Series
- Documentation
As such, for this Sprint, we're thinking of having two people set on figuring out how to work with and integrate PDS4, one person working on Wind Time, and the fourth working on the documentation, which we are thinking we'll make a simple webpage for, because as it stands right now, the Jupyter files in the project require multiple dependencies installed all just to view the READMEs and such, which is inefficient. 

Key notes:

- Sena and Jim will take the PDS4 integration/comprehension
- Sam will continue her Time Series work with the Wind element
- Lukas will take the webpage for documentation
- The team has elected to always be in contact while working, and scheduled usual class times for Tue/Thur, as well as 14:00 Saturdays

New Roles:
- Sena: Product Owner
- Sam: Dev
- Jim: Scribe
- Lukas: Project Lead

**Comment by JimmyCIsCrazySee** [comment]
10/04/22 - Sprint 2 Part 2
All present

Summary:
Midterms are starting sporadically, so team members are looking at an asymmetrical workload
Some members will have to reshuffle their scheduling, but all committed to getting work done by usual due dates (i.e., this Saturday, within the group)

Progress and Notes:
PDS4: We now know the module exists, and we've managed to call commands, but are still not getting consistent results from using it. Not sure how much of it is underdevelopment (Planetary Data Standard 4 Tools does not seem to have a great deal of use in the community) and how much of it is unfamiliarity. We (Sena and Jim) have some plans in place, as well as a couple of tutorials that we intend to examine this week, but due to the additional lack of definition from the sponsor in terms of how to use PDS4, we are raising the potential topic of an alternative, should the need arrive.

Webpage development: Currently working with dependencies for programs like Django in order to set up and render the webpage for documentation and information. As an alternative, we may go with a basic hard-code HTML solution, in order to minimize third-party programs, add-ons, and dependency risks (which compound with scale). Options are kept open here, but confidence of completion is at an all-time high.

Wind Time Series: Currently translating equations from White Paper into Python methods/functions. Also constructing unit tests. Progress continues apace, not too many alternatives or sticking points to be worried about here.

Currently we are still interested in a meeting with the sponsor to get a couple questions answered and figure out which of the above alternatives, where appropriate, we will be going with. While he is gone for this week, we will continue to explore these options on our own and keep contact with Hallie, our secondary contact.

**Comment by JimmyCIsCrazySee** [comment]
10/13/22 - Sprint 2 Retro
All present

PDS4 Integration - 
     Successfully imported, running, is capable of reading remote files; local readability giving minor trouble
     Cannot currently continue without the actual files that are germane to this project; random files found on the internet will not suffice
     Moving into Sprint 3, we are aiming to implement methods to process wind speed, pressure, and temperature data in this project
     Most pressing issue is to get a meeting with sponsor in order to get appropriately-PDS4-labeled data around which to build

Webpage and Documentation - 
    Webpage constructed, background philosophy and requirements are explained therein
    Linked Github repository as well as sponsor's website, in addition to relevant publications related to research, in order to 
          centralize as much information as conveniently as possible
    Intending to add specific examples of data from project in order to be able to expound upon with a little more specificity
    Driving philosophy is to provide a contrast between the relatively-more-arcane Jupyter files, making a format that's much more
          beginner-user-friendly
     
Wind Time Series - 
     Implemented functionality for equations and functions relevant to wind speed profile of a given vortex
     Implemented functions have been tested with PyTest, with promising results
     Working on modeling capability with wind speed time series
     As certain as can possibly be that required functions and methods are included and successfully implemented, but require input from
          sponsor in order to confirm. Additionally, require sponsor-direct data that is available in order to make sure the functions we're 
          building are accurate as well as precise.

New Roles:
Sena: Dev
Sam: Project Lead
Jim: Scribe
Lukas: Product Owner

**Comment by JimmyCIsCrazySee** [comment]
10/18/22 Sprint 3 Meeting 2
All present

PDS4 Integration -
Currently can read PDS4 data.  Read_data.py has been modified to fork based on input file format; it can read CSV files the same way it could before being handed off to us, but now we also have a functionality to detect and process PDS4 files. This needs to be applied across several different files and functions, and not every one of them is done yet, but the general process is the same, meaning applying it in other areas of the code is easier/quicker. Because of the nature of this change (that is, forking the program based entirely on what format is being supplied to the program), as well as the sheer volume of global variables and module imports, we have also elected not to refactor the entire project as we had discussed a week prior.

API and Data Processing - 
FastAPI has the ability to read any source file off our data git supplied by Dr. Jackson and rapidly render graphs and output. Considering adding this to our project. It is flexible enough that the API allows a user to specify even the exact day's data they would like to use, and is quite convenient.

Wind Time Series -
New Jupyter notebook created in order to be able to test with impunity. Have begun creating plots and figures that Dr. Jackson had originally created for pressure time series, in order to maintain consistency. Originally there was no read function for wind data, so that function has been created in the image of analogous functions (sans PDS4 function, for now).  These have since been committed and will be merged into the main branch soon.

**Comment by JimmyCIsCrazySee** [comment]
10/25/22 - Sprint 3 Retro/Sprint 4 Init
All present

What Worked:
PDS4 Integ - Pressure data for XML and CSV formats is now integrated and working successfully, by all measures
                      This part of the project seems to be done, as of now, with the exception of documents/PyDocs

Wind Speed - Despite Jupyter Notebooks not being available, wind time series is as close to fully implemented as it can be, until Dr. 
                          Jackson creates some missing functions
                       Unit tests for utilty functions are working
                       Created new function to obtain float arrays of time and object data, as its own object, which can be passed to other parts of 
                          the program

API - API is completed, and speed has been refined.
         A few pre-existing functions in the JN have also been tweaked to resolve faster and have more error handling
         Currently running at "90%" functionality
         

What Didn't:
JN was bugged for about two days, slowing us down, but has since been rectified 
We remain incapable of committing directly to the public Muldoon package Main branch, which makes it much harder for us to test our developments. We met with Dr. Jackson today and told him as much, and despite being added to the project, we still don't have that ability, and that means that if we try to run tests on remote machines, when they import Muldoon, they import the official package release, which is Dr. Jackson's version, which does not have our dev's on it. A specific example of an issue this causes us is that we can't be sure that our remote executions are producing graphic content when we expect them to.


Sprint 4 Goals:
- Focus on parameterization of program (much of it is still hardcoded and we need to get rid of that)
- Documentation (especially for read_data PDS4 functions, as above)
- Create script to validate, verify, and demonstrate different subsections of program
- Folding into above: begin composing demo  


New Roles:
Sena: Project Lead
Sam: Product Owner
Jim: Scribe
Lukas: Dev

**Comment by JimmyCIsCrazySee** [comment]
11/01/22 - Sprint 4 Part 2
All present

Summary
Work has been slow as yet; we all have other works to worry about and this one is going quite well so far. We have had our meeting with Dr. Jackson and appear to have circumvented our problems with Jupyter Notebooks being unusable, and now we can make use of our changes to the Muldoon codebase, demonstrably so.

Next steps are to hopefully consolidate all graphic and plotting functions into a single algorithm which is called by different sectors of the program depending on what they need to plot. We hope to close S4 in so doing. S5's focus is currently intended to be creation of the demo for our sponsor, as well as general wrap-up for the program.

**Comment by JimmyCIsCrazySee** [comment]
11/07/22 - Sprint 4 Retro/Sprint 5 Init
All present

Summary
We have been tweaking some of our new functions that we've written to streamline the program; some input files are quite large and we do not usually need to process the entire file. We will continue to trim down the efficiency of our algorithms to make execution as simple (we're consolidating functions where possible so users don't have to learn as much in order to use Muldoon effectively) as possible.

Jupyter Notebook demos are proceeding as needed. Wind is almost done, Temperature is underway, and we are about to start Pressure. These will be a focus of Sprint 5.

API is now capable of reading any directory in the project from the website, although we have noticed some erroneous returns on plotting pressure data graphs. It is currently unclear if this was due to tester error or if testing happened to use an outlier input file which did not have data in the column the API was attempting to examine.

Going Forward
Sprint 5's focus is to finish the Notebooks, which will segue neatly into creating our demonstration video. We'd also like to see about getting some error-handling inserted into the program (to handle situations like the above "this column might be empty" ambiguities). Further, we would like to wrap up and cap what we're doing with the webpage, while leaving it open for future development; we feel that the amount of work done at this point on that part of the project is sufficient and we would like to focus elsewhere.
We intend also to get our demo video finished by end of Sprint 5.

In terms of the merging/updating repo issue, we are considering the idea of forking the repository over again, and then just moving over our entire current-day git over it and PR'ing it to him. While that would be a considerable changelog for a single merge, we may simply ask him to archive his project, everything before Fall of 2022, and then we can hand him everything we've done all at once.

All in all, we are thinking we might line up another meeting with him, a longer one, where we can float this idea to him, show him everything we've done, outline our Sprint plans (Notebooks, demo, signature, handoff, etc), and that should neatly set us up on a path to go from here to having every box checked, the project handed off, and the class passed by semester's end.

New Roles
Lukas - Project Lead
Jim - Product Owner
Sena - Dev
Sam - Scribe

**Comment by samfarmer1** [comment]
11/15/22 - Sprint 5 Part 2
All present

Summary
We've been wrapping up our project and are now focused on giving a proper demo to our Sponsor on 11/16. We went over how to prepare for our demo, using Jupyter Notebooks, and the order we will present. We plan to discuss a deadline for which we would like him to sign off on our project and when we should officially hand off our project, at the end of the meeting.

Next steps are to work on a PowerPoint presentation for our Pitch Video, setting up our GitHub Page, and Drafting our project handoff document. We plan to meet this weekend to collaborate on these deliverables.

**Comment by samfarmer1** [comment]
11/29/22 - Sprint 5 Retro
All present

Summary
Sena and Jim - fixed bugs in the read_data.py script and added functions into the Using_Muldoon Jupyter Notebooks. They revised the Using_Muldoon notebook and we will be using that as our main demo/example in our presentation and pitch video.
Lukas - fixed bugs in the API, implemented the Wind Speed series and now there is full functionality for the files on the MEDA website.
Sam - created a function in the read_data script to improve the plots produced, cleaned up some old branches, created and launched the github page (still has too be finalized).

Next Steps:
We are now working on our final deliverables to the sponsor and plan to meet this weekend to work through everything. Final deliverables include: GitHub page, pitch video, handoff document, and slides for pitch video and final presentation.

Roles
Lukas - Project Lead
Jim - Product Owner
Sena - Dev
Sam - Scribe
